kura/examples/org.eclipse.kura.example.wire.math.singleport.provider/src/main/java/org/eclipse/kura/example/wire/math/singleport/RunningExtremes.java

```diff
@@ -0,0 +1,64 @@
+package org.eclipse.kura.example.wire.math.singleport;
+
+import java.util.ArrayDeque;
+
+public class RunningExtremes {
+
+    private final ArrayDeque<Double> window;
+    private final int windowSize;
+
+    private double max;
+    private double min;
+    private double last;
```

Comment:
The usage of "last" has been thought in order to reduce computation for this component.

The value being extracted from the array is checked in order to understand if it was either the maximum or the minimum; if it is not, there is no need to update these two values, which can then be checked against the new value being inserted in the array.
Assuming my maximum in the window is 5 and my minimum is 1, if I extract a value of 2, I can still keep those two variables without re-check the whole window. Then I can check the new value against reliable local maximums and minimums. I reckon with a window of quite a lot of elements, and a quick running Wire Graph application, this can help.

It is of course feasible to implement this without this check, but at each iteration the component will have to check the whole array, which is pretty much a waste of resources.